packages/headless/src/modules/brands/services/index.ts

```diff
@@ -0,0 +1,61 @@
+
+import { HttpStatus, Injectable } from "@nestjs/common";
+import { Response } from "express";
+import * as crypto from 'crypto';
+import { ErrorMessage, ErrorMessageDeleteBrand, ErrorMessageGetBrandById, ErrorMessageUpdate } from "models";
+
+import { BrandCreateSchema } from './../validators/brand.create.validator';
+import { Brand } from 'src/entity/brand';
+import { BrandRepository } from './../repositories/index';
+import { CreateBrandRequestDto, CreateBrandResponseDto } from 'src/modules/brands/dto/createBrandDto';
+import { GetAllBrandsResponseDto } from "src/modules/brands/dto/getAllBrandsDto";
+import { UpdateBrandRequestdto, UpdateBrandResponseDto } from "src/modules/brands/dto/updateBrandDto";
+import { BrandDto } from 'src/modules/brands/dto/brandDto';
+import { GetBrandByIdResponseDto } from "src/modules/brands/dto/getBrandByIdDto";
+import { DeleteBrandResponseDto } from "src/modules/brands/dto/deleteBrandDto";
+
+@Injectable()
+
+export class BrandService{
+    constructor( private brandRepo: BrandRepository ){}
+
+    async createBrand( brand: CreateBrandRequestDto ): Promise<CreateBrandResponseDto>{
+            const doesBrandExist = await this.brandRepo.getBrandByName(brand.info.name);
+            if(doesBrandExist) return { error: ErrorMessage.BRAND_ALREADY_EXISTS, errors: null, code: HttpStatus.BAD_REQUEST};
+            else {
+                const newBrand = await this.brandRepo.createBrand(brand);
+                if (!newBrand) return { error: ErrorMessage.CANNOT_CREATE_BRAND, errors: null, code: HttpStatus.INTERNAL_SERVER_ERROR };
+                 
+                return { code: HttpStatus.CREATED, data: newBrand }; 
+            }    
+    }
+    
+    async getAllBrands(skip?: number, limit?: number): Promise<GetAllBrandsResponseDto>{
+        const allBrands = await this.brandRepo.getAllBrands(skip, limit);
+        if(!allBrands) return { error: 'Could not get brands due to server error', errors: null, code: HttpStatus.INTERNAL_SERVER_ERROR };
```

Comment:
why hard coded error

Reply:
hard coded because the error while getting all brands is only the internal server error, thats why didnot create an enum for a single error message